lib/_stream_readable.js

```diff
@@ -1007,83 +1008,53 @@ function flow(stream) {
 // This is *not* part of the readable stream interface.
 // It is an ugly unfortunate mess of history.
 Readable.prototype.wrap = function(stream) {
-  const state = this._readableState;
-  let paused = false;
+  let paused;
 
-  stream.on('end', () => {
-    debug('wrapped end');
-    if (state.decoder && !state.ended) {
-      const chunk = state.decoder.end();
-      if (chunk && chunk.length)
-        this.push(chunk);
-    }
-
-    this.push(null);
-  });
+  // TODO (ronag): Should this.destroy(err) emit
+  // 'error' on the wrapped stream? Would require
+  // a static factory method, e.g. Readable.wrap(stream).
 
   stream.on('data', (chunk) => {
-    debug('wrapped data');
-    if (state.decoder)
-      chunk = state.decoder.write(chunk);
-
-    // Don't skip over falsy values in objectMode.
-    if (state.objectMode && (chunk === null || chunk === undefined))
-      return;
-    else if (!state.objectMode && (!chunk || !chunk.length))
-      return;
-
-    const ret = this.push(chunk);
-    if (!ret) {
+    if (!this.push(chunk) && stream.pause) {
       paused = true;
       stream.pause();
     }
   });
 
-  // Proxy all the other methods. Important when wrapping filters and duplexes.
-  for (const i in stream) {
-    if (this[i] === undefined && typeof stream[i] === 'function') {
-      this[i] = function methodWrap(method) {
-        return function methodWrapReturnFunction() {
-          return stream[method].apply(stream, arguments);
-        };
-      }(i);
-    }
-  }
+  stream.on('end', () => {
+    this.push(null);
+  });
 
   stream.on('error', (err) => {
     errorOrDestroy(this, err);
   });
 
   stream.on('close', () => {
-    // TODO(ronag): Update readable state?
-    this.emit('close');
+    this.destroy();
   });
 
   stream.on('destroy', () => {
-    // TODO(ronag): this.destroy()?
-    this.emit('destroy');
+    this.destroy();
   });
 
-  stream.on('pause', () => {
-    // TODO(ronag): this.pause()?
-    this.emit('pause');
-  });
-
-  stream.on('resume', () => {
-    // TODO(ronag): this.resume()?
-    this.emit('resume');
-  });
```

Comment:
I'm a little unsure what to do exactly when the wrapped stream emits pause and resume. I don't think it's relevant?